src/NextGen/identity/identity.ts

```diff
@@ -1,48 +1,134 @@
 import { MPID, User } from '@mparticle/web-sdk';
 import { IMParticleComponent } from '../core/component';
+import { Context } from '@mparticle/event-models';
 
-// QUESTION: What is the diff between this and IdentityApiData?
-// TODO: sync this with android class
-export interface IdentityApiResult {}
-export interface IdentityApiRequest {}
+export interface IdentityApiResult {
+    user: User;
+    previousUser: User;
 
-// QUESTION: Should we mirror the way Android handles HTTP codes, or do it the way we've been doing it?
-// https://docs.mparticle.com/developers/sdk/android/idsync/#idsync-status-codes
-// https://docs.mparticle.com/developers/sdk/web/idsync/#error-handling
+    getUser(): User;
+    getPreviousUser(): User;
+}
+
+export enum MParticleServerErrorCodes {
+    noHttpCoverage = -1,
+    activeIdentityRequest = -2,
+    activeSession = -3,
+    validationIssue = -4,
+    nativeIdentityRequest = -5,
+    loggingDisabledOrMissingAPIKey = -6,
+    tooManyRequests = 429,
+}
+
+export interface IdentityAPIServerErrorResult {
+    errorCode: MParticleServerErrorCodes;
+}
+
+export interface IdentityAPIHttpErrorResult {
+    errorCode: number;
+}
+
+export type IMPIdentityApiResponse =
+    | IdentityApiResult
+    | IdentityAPIHttpErrorResult
+    | IdentityAPIServerErrorResult;
+
+type IdentityTypes =
+    | 'other'
+    | 'customerid'
+    | 'facebook'
+    | 'twitter'
+    | 'google'
+    | 'microsoft'
+    | 'yahoo'
+    | 'email'
+    | 'facebookcustomaudienceid'
+    | 'other2'
+    | 'other3'
+    | 'other4'
+    | 'other5'
+    | 'other6'
+    | 'other7'
+    | 'other8'
+    | 'other9'
+    | 'other10'
+    | 'phone_number_2'
+    | 'phone_number_3'
+    | 'mobile_number'
+    | 'device_application_stamp';
+
+type KnownIdentities = {
+    [key in IdentityTypes]?: string;
```

Comment:
This will actually always have a `device_application_stamp` on it at a minimum.  everything else is optional